src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -173,3 +173,110 @@ def watch():
         t = threading.Thread(target=watch, daemon=True)
         t.start()
         t.join()
+
+
+class ZFSSnapshot(Service):
```

Comment:
I think that is a great idea and is the right thing to do; but also I think it is out of the scope of this PR.

How about make these changes as a next step for plugins/zfs.py and sumbit another PR with these changes as well as including the changes in the OLD GUI?